### PR TITLE
Scenes: Fix panel repeats

### PR DESCRIPTION
**Pull request by kaydelaney**
Fixes an issue where panel repeats lose their data when entering/exiting panel editor

### CODE REVIEW

**Review by mdvictor — APPROVED** [review]

**Review by mdvictor — CHANGES_REQUESTED** [review]
Think this still needs some changes, I still managed to reproduce on a newly created dashboard with a panel that has repeat options set. Saving the dashboard and refreshing the browser fixes this, though

**Comment by kaydelaney** [comment]
Hopefully fixed the other failures with that last commit, let me know if it works for you @mdvictor

File: public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ export class PanelEditor extends SceneObjectBase<PanelEditorState> {
       width: repeatDirection === 'h' ? 24 : gridItem.state.width,
       height: gridItem.state.height,
       itemHeight: gridItem.state.height,
-      source: panelManager.state.panel.clone(),
+      source: panelManager.state.panel.clone({ $data: panelManager.state.$data?.clone() }),
```
Comment (torkelo) [inline public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx:141]: maybe we can extract this to function on VizPanelManager as it is repeated  so many times

**Review by torkelo — APPROVED** [review]
would be nice with a unit test , looks like no existing tests for committing changes to repeated panel though :/

File: public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx (hunk 1 above)
Comment (kaydelaney) [reply public/app/features/dashboard-scene/panel-edit/PanelEditor.tsx:141]: Good idea

File: public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx [hunk 2]
```diff
@@ -367,6 +367,10 @@ export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
     return { error: 'Unsupported panel parent' };
   }
 
+  public clonePanelWithData = () => {
```
Comment (torkelo) [inline public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx:370]: why arrow function?

Comment (kaydelaney) [reply public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx:370]: Just habit 😓 Changed

Comment (torkelo) [reply public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx]:
```suggestion
  public getPanelCloneWithData {
```

**Review by mdvictor — APPROVED** [review]
Awesome work, thank you @kaydelaney! 

I found another bug but this looks like it's related to the grid layout keys. Will open a new [bug report](https://github.com/grafana/grafana/issues/83827) for this.


https://github.com/grafana/grafana/assets/36818606/f611f996-f2a3-4c4c-b265-03b3da13a407

**Pull request merged**